### ISSUE TITLE
`Mac_arm64_staging plugin_lint_mac` failed with xcodebuild failure

### ISSUE THREAD

**Issue by keyonghan** [flutter/flutter#102108]
Failed build: https://luci-milo.appspot.com/raw/build/logs.chromium.org/flutter/led/keyonghan_google.com/d98f95cc604bc76f37631187747084638ef76fcf9b65ea178741cd88a9b8b462/+/build.proto
(This is with a customized arm64 Ruby).

```
[plugin_lint_mac] [STDOUT] stdout:         write-file /Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Intermediates.noindex/App.build/Debug/App.build/App-generated-files.hmap
[plugin_lint_mac] [STDOUT] stdout:     
[plugin_lint_mac] [STDOUT] stdout:     WriteAuxiliaryFile /Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Intermediates.noindex/App.build/Debug/App.build/App-all-target-headers.hmap (in target 'App' from project 'App')
[plugin_lint_mac] [STDOUT] stdout:         cd /opt/s/w/ir/x/t/CocoaPods-Lint-20220418-6248-ru11t2-integration_test_macos
[plugin_lint_mac] [STDOUT] stderr:     ** BUILD FAILED **
[plugin_lint_mac] [STDOUT] stderr:     
[plugin_lint_mac] [STDOUT] stderr:     
[plugin_lint_mac] [STDOUT] stderr:     The following build commands failed:
[plugin_lint_mac] [STDOUT] stderr:     	Ld /Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Products/Debug/integration_test_macos/integration_test_macos.framework/Versions/A/integration_test_macos normal (in target 'integration_test_macos' from project 'Pods')
[plugin_lint_mac] [STDOUT] stderr:     (1 failure)
[plugin_lint_mac] [STDOUT] stdout:         write-file /Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Intermediates.noindex/App.build/Debug/App.build/App-all-target-headers.hmap
[plugin_lint_mac] [STDOUT] stdout:     
[plugin_lint_mac] [STDOUT] stdout:     WriteAuxiliaryFile /Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Intermediates.noindex/App.build/Debug/App.build/App-all-non-framework-target-headers.hmap (in target 'App' from project 'App')
[plugin_lint_mac] [STDOUT] stdout:         cd /opt/s/w/ir/x/t/CocoaPods-Lint-20220418-6248-ru11t2-integration_test_macos
[plugin_lint_mac] [STDOUT] stdout:         write-file /Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Intermediates.noindex/App.build/Debug/App.build/App-all-non-framework-target-headers.hmap
[plugin_lint_mac] [STDOUT] stdout:     
[plugin_lint_mac] [STDOUT] stdout:     Ld /Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Products/Debug/integration_test_macos/integration_test_macos.framework/Versions/A/integration_test_macos normal (in target 'integration_test_macos' from project 'Pods')
[plugin_lint_mac] [STDOUT] stdout:         cd /opt/s/w/ir/x/t/CocoaPods-Lint-20220418-6248-ru11t2-integration_test_macos/Pods
[plugin_lint_mac] [STDOUT] stdout:         /opt/flutter/xcode/13a233/XCode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target arm64-apple-macos10.11 -dynamiclib -isysroot /opt/flutter/xcode/13a233/XCode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk -L/Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Products/Debug/integration_test_macos -L/opt/flutter/xcode/13a233/XCode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx -L/opt/flutter/xcode/13a233/XCode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/usr/lib/swift -F/Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Products/Debug/integration_test_macos -F/opt/s/w/ir/x/t/CocoaPods-Lint-20220418-6248-ru11t2-integration_test_macos/Pods/FlutterMacOS -filelist /Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Intermediates.noindex/Pods.build/Debug/integration_test_macos.build/Objects-normal/arm64/integration_test_macos.LinkFileList -install_name @rpath/integration_test_macos.framework/Versions/A/integration_test_macos -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker @executable_path/../Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -Xlinker -object_path_lto -Xlinker /Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Intermediates.noindex/Pods.build/Debug/integration_test_macos.build/Objects-normal/arm64/integration_test_macos_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -fobjc-arc -fobjc-link-runtime -L/opt/flutter/xcode/13a233/XCode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Intermediates.noindex/Pods.build/Debug/integration_test_macos.build/Objects-normal/arm64/integration_test_macos.swiftmodule -framework FlutterMacOS -framework Cocoa -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Intermediates.noindex/Pods.build/Debug/integration_test_macos.build/Objects-normal/arm64/integration_test_macos_dependency_info.dat -o /Users/swarming/Library/Developer/Xcode/DerivedData/App-fmkgmwxvljdjzpaoqkmjmonjccpb/Build/Products/Debug/integration_test_macos/integration_test_macos.framework/Versions/A/integration_test_macos
[plugin_lint_mac] [STDOUT] stdout:     ld: warning: ignoring file /opt/s/w/ir/x/t/CocoaPods-Lint-20220418-6248-ru11t2-integration_test_macos/Pods/FlutterMacOS/FlutterMacOS.framework/FlutterMacOS, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
[plugin_lint_mac] [STDOUT] stdout:     Undefined symbols for architecture arm64:
[plugin_lint_mac] [STDOUT] stdout:       "_FlutterMethodNotImplemented", referenced from:
[plugin_lint_mac] [STDOUT] stdout:           integration_test_macos.IntegrationTestPlugin.handle(_: __C.FlutterMethodCall, result: (Any?) -> ()) -> () in IntegrationTestPlugin.o
[plugin_lint_mac] [STDOUT] stdout:       "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
[plugin_lint_mac] [STDOUT] stdout:           objc-class-ref in IntegrationTestPlugin.o
[plugin_lint_mac] [STDOUT] stdout:     ld: symbol(s) not found for architecture arm64
[plugin_lint_mac] [STDOUT] stdout:     clang: error: linker command failed with exit code 1 (use -v to see invocation)
[plugin_lint_mac] [STDOUT] stdout:     
[plugin_lint_mac] [STDOUT] stdout:   Testing with `xcodebuild`.
[plugin_lint_mac] [STDOUT] stdout:  -> integration_test_macos (0.0.1)
[plugin_lint_mac] [STDOUT] stdout:     - ERROR | [OSX] xcodebuild: Returned an unsuccessful exit code.
[plugin_lint_mac] [STDOUT] stdout:     - NOTE  | xcodebuild:  note: Using new build system
[plugin_lint_mac] [STDOUT] stdout:     - NOTE  | xcodebuild:  note: Using codesigning identity override: 
[plugin_lint_mac] [STDOUT] stdout:     - NOTE  | xcodebuild:  note: Build preparation complete
[plugin_lint_mac] [STDOUT] stdout:     - NOTE  | [OSX] xcodebuild:  note: Planning
[plugin_lint_mac] [STDOUT] stdout:     - NOTE  | [OSX] xcodebuild:  note: Building targets in parallel
[plugin_lint_mac] [STDOUT] stdout:     - NOTE  | [OSX] xcodebuild:  ld: warning: ignoring file FlutterMacOS/FlutterMacOS.framework/FlutterMacOS, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
[plugin_lint_mac] [STDOUT] stdout:     - NOTE  | [OSX] xcodebuild:  clang: error: linker command failed with exit code 1 (use -v to see invocation)
[plugin_lint_mac] [STDOUT] stdout: 
[plugin_lint_mac] [STDOUT] stdout: [!] integration_test_macos did not pass validation, due to 1 error.
[plugin_lint_mac] [STDOUT] stdout: You can use the `--no-clean` option to inspect any issue.
[plugin_lint_mac] [STDOUT] stdout: 
[plugin_lint_mac] [STDOUT] stdout: /opt/s/w/ir/x/w/gems/ruby/3.1.0/gems/cocoapods-1.11.3/lib/cocoapods/command/lib/lint.rb:113:in `block in run'
[plugin_lint_mac] [STDOUT] stdout: /opt/s/w/ir/x/w/gems/ruby/3.1.0/gems/cocoapods-1.11.3/lib/cocoapods/command/lib/lint.rb:75:in `each'
[plugin_lint_mac] [STDOUT] stdout: /opt/s/w/ir/x/w/gems/ruby/3.1.0/gems/cocoapods-1.11.3/lib/cocoapods/command/lib/lint.rb:75:in `run'
[plugin_lint_mac] [STDOUT] stdout: /opt/s/w/ir/x/w/gems/ruby/3.1.0/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
[plugin_lint_mac] [STDOUT] stdout: /opt/s/w/ir/x/w/gems/ruby/3.1.0/gems/cocoapods-1.11.3/lib/cocoapods/command.rb:52:in `run'
[plugin_lint_mac] [STDOUT] stdout: /opt/s/w/ir/x/w/gems/ruby/3.1.0/gems/cocoapods-1.11.3/bin/pod:55:in `<top (required)>'
[plugin_lint_mac] [STDOUT] stdout: /opt/s/w/ir/x/w/gems/ruby/3.1.0/bin/pod:25:in `load'
[plugin_lint_mac] [STDOUT] stdout: /opt/s/w/ir/x/w/gems/ruby/3.1.0/bin/pod:25:in `<main>'
```

**Comment by jmagman** [comment]
Duplicate of https://github.com/flutter/flutter/issues/100638
